proxy/airdropper/airdropper.py

```diff
@@ -199,6 +236,98 @@ def airdrop_to(self, eth_address, airdrop_galans):
+        LOG.debug(f"from: {sender}, to: {to}, calldata: {trx.callData.hex()}")
+
+        analyzer = self.analyzers.get(to, None)
+        if analyzer is not None:
+            try:
+                analyzer.process(trx, self)
+            except Exception as error:
+                LOG.warn(f"Failed to analyze {sol_neon_ix.neon_tx_sig}: {error}")
+
+    # Method to process Solana transactions and extract NeonEVM transaction from the contract instructions.
+    # For large NeonEVM transaction that passing to contract via account data, this method extracts and 
+    # combines chunk of data from different HolderWrite instructions. At the any time `neon_large_tx` 
+    # dictionary contains actual NeonEVM transactions written into the holder accounts. The stored account 
+    # are cleared in case of execution, cancel trx or writing chunk of data from another NeonEVM transaction. 
+    # This logic are implemented according to the work with holder account inside contract.
+    # Note: the `neon_large_tx` dictionary stored only in memory, so `last_processed_slot` move forward only 
+    # after finalize corresponding holder account. It is necessary for correct transaction processing after 
+    # restart the airdriop service.
+    # Note: this implementation analyzes only the final step in case of iterative execution. It simplifies it 
+    # but does not process events generated from the Solidity contract.
+    def process_trx_neon_instructions(self, trx):
+        if check_error(trx):
+            return
+
+        tx_receipt_info = SolTxReceiptInfo.from_tx(trx)
+        sol_sig = tx_receipt_info.sol_sig
+        for sol_neon_ix in tx_receipt_info.iter_sol_neon_ix():
+            instruction = sol_neon_ix.ix_data[0]
+            neon_tx_sig = NeonIndexedHolderInfo.Key("", sol_neon_ix.neon_tx_sig).neon_tx_sig
+            LOG.debug(f"{sol_sig} instruction: {instruction} {sol_neon_ix.neon_tx_sig}")
+            if instruction == EVM_LOADER_HOLDER_WRITE:
+                holder = SolPubKey.from_string(sol_neon_ix.get_account(0))
+                data = sol_neon_ix.ix_data[41:]
+                chunk = NeonIndexedHolderInfo.DataChunk(
+                    offset=int.from_bytes(sol_neon_ix.ix_data[33:41],'little'),
+                    length=len(data),
+                    data=data
+                )
+                neon_tx_data = self.neon_large_tx.get(holder, None)
+                if neon_tx_data and neon_tx_data.neon_tx_sig == neon_tx_sig:
+                    LOG.debug(f"{sol_sig} Add data to NEON trx: {holder} {len(chunk.data)} bytes at {chunk.offset}")
+                else:
+                    LOG.debug(f"{sol_sig} New NEON trx: {holder} {len(chunk.data)} bytes at {chunk.offset}")
+                    key = NeonIndexedHolderInfo.Key(sol_neon_ix.get_account(0), neon_tx_sig)
+                    neon_tx_data = NeonIndexedHolderInfo(key)
+                    self.neon_large_tx[holder] = neon_tx_data
```

Comment:
`self.neon_large_tx[key.value] = neon_tx_data`

Reply:
Reworked to HOLDER:NEON_TX_SIG as key